lib/src/features/game/domain/game.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:dartchess/dartchess.dart';
+
+part 'game.freezed.dart';
+part 'game.g.dart';
+
+@Freezed(toJson: false)
+class Game with _$Game {
+  factory Game({
+    required String id,
+    required bool rated,
+    required Speed speed,
+    required String initialFen,
```

Comment:
a `Fen` type would be nice. In scalachess I even have several types based on the fen content: 

```scala
// r3k2r/p3n1pp/2q2p2/4n1B1/5Q2/5P2/PP3P1P/R4RK1 b kq - 6 20
opaque type Fen = String

// r3k2r/p3n1pp/2q2p2/4n1B1/5Q2/5P2/PP3P1P/R4RK1
opaque type BoardFen = String
```


https://github.com/lichess-org/scalachess/blob/master/src/main/scala/format/Fen.scala

Reply:
this new `opaque type` feature of scala3 looks very cool indeed.

Reply:
here we could do `typedef Fen = String` but that's not as much useful...